components/scream/src/physics/shoc/shoc_assumed_pdf_impl.hpp

```diff
@@ -146,17 +146,19 @@ void Functions<S,D>::shoc_assumed_pdf(
         Skew_thl.set(tsign>sp(0.2) && tsign<=sp(0.4), (((sp(1.2)*Skew_w)/sp(0.2))*(tsign-sp(0.2))));
       }
 
-      thl2_1.set(condition,
-                 ekat::min(100,
-                           ekat::max(0, (3*tmp_val_1*(1 - a*ekat::square(tmp_val_2) - (1-a)*ekat::square(tmp_val_1))
-                                         - (Skew_thl - a*ekat::cube(tmp_val_2) - (1 - a)*ekat::cube(tmp_val_1)))
-                                         /(3*a*(tmp_val_1 - tmp_val_2))))*thlsec);
-      thl2_2.set(condition,
-                 ekat::min(100,
+      if (condition.any()) {
+        thl2_1.set(condition,
+                   ekat::min(100,
+                             ekat::max(0, (3*tmp_val_1*(1 - a*ekat::square(tmp_val_2) - (1-a)*ekat::square(tmp_val_1))
+                                           - (Skew_thl - a*ekat::cube(tmp_val_2) - (1 - a)*ekat::cube(tmp_val_1)))
+                                       /(3*a*(tmp_val_1 - tmp_val_2))))*thlsec);
+        thl2_2.set(condition,
+                   ekat::min(100,
                            ekat::max(0, (-3*tmp_val_2*(1 - a*ekat::square(tmp_val_2)
                                          - (1 - a)*ekat::square(tmp_val_1))
                                          + (Skew_thl - a*ekat::cube(tmp_val_2) - (1 - a)*ekat::cube(tmp_val_1)))
                                          /(3*(1 - a)*(tmp_val_1 - tmp_val_2))))*thlsec);
+      }
```

Comment:
Looks like there is the same opportunity here to do one big `condition.any()` check to either run or not run a big chunk of code. That will then naturally solve the /0 problem.